fix: Deltoid wrapping

Hamedshayestehpour opened the pull request:
New ellipsoids were defined based on the nodes on the segments instead of fixed angles so it can scale properly.
Constraints were redefined to use nodes on the segments instead of rotational rhythms, improved when scaled and even on unscaled wrapping.
A cylinder were defined on the humerus which allow the deltoids to be able to make their functions as external and interal rotation of the humerus

Overall, the wrapping is improved and in the ROM with no penetrations, which happened in some postures before.

Commit:
Renaming muscle groups and sorting muscles alphabetically

Commit:
description for longissimus thoracis

Commit:
Fix part of illiocostalis and longissimus names of muscle model reference (model does not load)

Commit:
fix evaluate moment arm study names of illiocostalis and longissimus (model does not load)

Commit:
muscle parameter names fix, sorted alphabetically (model does not load)

Commit:
Muscle parameters (first draft completed)

Commit:
fix muscle view transversus throacis

Commit:
Diaphragm muscle naming

Commit:
fix bug in naming muscle parameters

Commit:
Iliocostalis muscle naming

Commit:
Intercostalis muscle name. Fixed bug, wrong muscle parameters used

Commit:
LevatorCostarum muscle name

Commit:
LongissimusCervicis muscle name

Commit:
LongissimusThoracis muscle name

Commit:
LongusColli and LongusCapitis muscle name

Commit:
MultifidusLumborum

Commit:
MultifidusThoracis

Commit:
ObliquusExternus Volume mode

Commit:
ObliquusExternus Buckle mode

Commit:
ObliquusInternus volume mode

Commit:
ObliquusInternus Buckle mode

Commit:
Sorting changes a bit due to changing muscle group names

Commit:
Puborectalis and Pupococcygeus

Commit:
QuadratusLumborum

Commit:
RectusAbdominis

Commit:
RotatoresLongus and Brevis

Commit:
Scalenus

Commit:
Scalenus

Commit:
SemispinalisThoracis

Commit:
SerratusPosteriorSuperior and SerratusPosteriorInferior

Commit:
Spinalis

Commit:
SpleniusCapitis

Commit:
Subcostalis

Commit:
TransversusAbdominis Volume

Commit:
TransversusThoracis

Commit:
PsoasMajor including changes in trunk and leg files

Commit:
Update and fix bug in muscle parameter files

Commit:
Fix typo LongusColli

Commit:
Fix bug in MultifidusLumborum

Revert "MultifidusLumborum"

This reverts commit 99c921d254da05a8d576f35edb6c6fbff85ea129.

Commit:
Muscle parameter update

Commit:
update muscle parameters

Commit:
fix Hyoid

Commit:
fix bugs

Commit:
fix bugs

Commit:
Bug fix

Commit:
Spelling fix

Commit:
Multifidi to MultifidusLumborum

Commit:
fix

Commit:
Fix issues

Commit:
Excluding Puborectalis and Pubococcygeus from being calibrated

Commit:
Fix test_calibration_trunk cervical motion avoiding zero in Lf0s of cervical muscles

Commit:
fix bugs

Commit:
Merge branch 'Muscl_Naming' into ArmMuscleNaming

Commit:
LevatorScapulae

Commit:
TrapeziusScapular

Commit:
TrapeziusClavicular

Commit:
Separate BicepsBrachiiBreve and Longum and Coracobrachialis naming

Commit:
DeltoideusPosterior

Commit:
DeltoideusLateral

Commit:
DeltoideusAnterior

Commit:
Infraspinatus

Commit:
LatissimusDorsi

Commit:
PectoralisMajorSternocostal

Commit:
PectoralisMajorClavicular

Commit:
PectoralisMinor

Commit:
Rhomboideus

Commit:
SerratusAnterior

Commit:
Subscapularis, Supraspinatus, TeresMajor, TeresMinor

Commit:
Brachialis

Commit:
TricepsBrachii Longum, Mediale, Laterale

Commit:
Brachioradialis, Anconeus, PronatorTeres

Commit:
Supinator

Commit:
PronatorQuadratus, ExtensorIndicis, AbductorPollicisLongus, ExtensorPollicis breve and longus, ExtensorCarpiRadialis breve and longus

Commit:
ExtensorCarpiUlnaris, FlexorCarpiUlnaris, FlexorCarpiRadialis, PalmarisLongus, FlexorDigitorumSuperficialis

Commit:
FlexorDigitorumProfundus, ExtensorDigitorum, ExtensorDigitiMinimi, FlexorPollicisLongus

Commit:
fix compatibility

Commit:
fix calibration (still need fix)

melund committed:
Merge branch 'ammr4-beta' into Muscl_Naming

Commit:
More fixes to muscle calibration

melund committed:
Merge branch 'ammr4-beta' into Muscl_Naming

Commit:
Fix calibration test

Commit:
Merge remote-tracking branch 'remotes/origin/ammr4-beta' into ArmMuscleNaming

Commit:
Merge remote-tracking branch 'remotes/origin/Muscl_Naming' into ArmMuscleNaming

Commit:
Remove Refs from muscle folder

Commit:
Moving references and wrapping surfaces from muscle folder to the WrappingSurfaces folder

Commit:
Defining folder in arm root folder

Commit:
fix bugs from extra folder

Commit:
Sort alphabetically

Commit:
fix name of PronatorTeresUlnare

Commit:
Muscle parameters ran by code

Hamedshayestehpour committed:
Merge branch 'ammr4-beta' into ArmMuscleNaming

Signed-off-by: Hamedshayestehpour <70632589+Hamedshayestehpour@users.noreply.github.com>

Commit:
removing unused nodes and fix typo in naming

Commit:
fix bug from code making muscle parameters

Commit:
Merge branch 'ArmMuscleNaming' of https://github.com/AnyBody/AMMR4-Beta into ArmMuscleNaming

Commit:
fix typo searching for a muscle

Commit:
Tweak the angles of the ellipsoids that wrap the surfaces of the posterior deltoid

Commit:
Tweak the angles of the ellipsoids that wrap the surfaces of the posterior deltoid

Commit:
Revert Tweaks

Commit:
fix typo

Commit:
sort deltoid elements

Commit:
fix for posterior deltoids (still in progress)

melund committed:
Run tests with AMS 8.2.0beta5

melund committed:
fix: update runner version to windows-2025 in workflow files

Commit:
continuing

Commit:
new ellipsoids and constraints are Avg measures 6DOF (not finished)

Commit:
New wrapping Works but needs clean up

Commit:
move the anterior ellipsoids to the anterior segment

Commit:
clean up and restructure

Commit:
clean up ellipsoid definitions

Commit:
Remove extra empty lines

Commit:
Merge remote-tracking branch 'remotes/origin/ammr4-beta' into DeltoidWrapping

Commit:
tuning the distances to avoid small penetration in upright arm posture

Commit:
Merge remote-tracking branch 'remotes/origin/test-with-ams-8.2.0beta5' into DeltoidWrapping

Commit:
improve deltoid wrapping

Commit:
SmallStepAssumptionOnOff to Off for tests

Commit:
update Deltoid wrapping

Commit:
change the order of wrapping surfaces

Commit:
Ellipsoid wrapping finished (works without penetration)

melund commented:
@Hamedshayestehpour, The large arm refactoring/inertia work made by @toerholm has been merged, so you may need to update this PR.

Commit:
Deltoid wrapping new setup
Anterior: on cyl moving with clavicula and gh joint. Have stop cylinders represent the lateral deltoid.
Lateral: on ellipsoids like the previous new versions
Posterior: on cyl moving with scapula and gh joint. Have stop cylinders represent the lateral deltoid.
All have humerus cyl, to wrapp on and create rotational moment in humerus

Commit:
Sphere and via nodes concept

Commit:
Via nodes, and cyl version for all

Commit:
unfinished version

Commit:
Clean up

Commit:
Move segments to lateral deltoid file

melund committed:
Merge branch 'ammr4-beta' into DeltoidWrapping

melund commented:
@toerholm. Did you take over this PR from @Hamedshayestehpour ?
